src/com/google/enterprise/adaptor/filenet/ConfigOptions.java

```diff
@@ -58,34 +65,46 @@ public Params(Config config) throws InvalidConfigurationException {
           "Invalid filenet.contentEngineUrl: " + e.getMessage());
     }
 
-    objectStore = config.getValue("filenet.objectStore").trim();
-    if (objectStore.isEmpty()) {
-      throw new InvalidConfigurationException("filenet.objectStore may not be empty");
+    objectStoreName = config.getValue("filenet.objectStore");
+    if (objectStoreName.isEmpty()) {
+      throw new InvalidConfigurationException(
+          "filenet.objectStore may not be empty");
     }
-    logger.log(Level.CONFIG, "filenet.objectStore: {0}", objectStore);
+    logger.log(Level.CONFIG, "filenet.objectStore: {0}", objectStoreName);
 
     username = config.getValue("filenet.username");
     password = config.getValue("filenet.password");
 
-    // TODO(bmj): load the object factor?
-    objectFactory = config.getValue("filenet.objectFactory").trim();
-    if (objectFactory.isEmpty()) {
+    String objectFactoryName = config.getValue("filenet.objectFactory");
+    if (objectFactoryName.isEmpty()) {
       throw new InvalidConfigurationException(
           "filenet.objectFactory may not be empty");
     }
-    logger.log(Level.CONFIG, "filenet.objectFactory: {0}", objectFactory);
+    try {
+      objectFactory =
+          (ObjectFactory) Class.forName(objectFactoryName).newInstance();
+    } catch (InstantiationException e) {
```

Comment:
I know this is a copy from FileSession, but the verbosity for errors that can't happen is too high. Let's just | these together.